### PR TITLE
[docs] general/projects/docs/migration - Detail devdocs migration project

### PR DESCRIPTION
**Pull request by andrewnicols**
This set of commits is a start on the documentation of the project, and the documentation of projects in general.

This includes:

- centralised project information in JSON
- schema validation of the JSON
- sidebar generation from the JSON
- an infobox-like display of the project summary from that JSON

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/96"></a>

### CODE REVIEW

File: general/projects/docs/migration.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+By hosting that repository in GitHub we can make use of GitHub actions to enable automatic documentation of certain features, including:
+
+- Web Service documentation
+- Lists of components, plugins, contributors, and more
+- Third-party libraries
+- upgrade.txt files distributed around Moodle
+
+## Migration plan
+
+Migration from the [Legacy DevDocs](https://docs.moodle.org/dev) comes in XXX parts:
```
Comment (sarjona) [inline general/projects/docs/migration.md:103]:
```suggestion
Migration from the [Legacy DevDocs](https://docs.moodle.org/dev) comes in three parts:
```

**Review by sarjona — CHANGES_REQUESTED** [review]
Hi Andrew!
I love this idea for implementing projects!! Thanks for working on it and creating the skeleton for migrating all the project pages!! It looks great and that way, it will be much more easier to find all these projects! <3

As you'll see, I've raised a couple of minor things (and apart from that, there is a conflict with the project-words.txt file).

File: src/components/ProjectSummary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                {owners}
+            </List>
+        );
+    }
+
+    function GetProjectSummary(projectData: ProjectSummaryData): ReactNode {
+        console.log(projectData);
+        return (
+            <table className={styles.projecttable}>
+                <caption>{projectData.title}</caption>
```
[the remark this answers is not in the document]
Comment (andrewnicols) [reply src/components/ProjectSummary/index.tsx:55]: agreed. I was in two minds about keeping it. Let's just get rid.

I've also removed that errant console.log

**Review by sarjona — APPROVED** [review]
Hi Andrew!
The patch looks great. I can't wait to see more projects (and also the MUI table! <3 ).

**Pull request merged**